Adding support for external authentication

bhoehl opened the pull request:
@pinepain @lstrojny 
The current implementation of php-amqp is using a static value for authentication in amqp_connection_resource.c function connection_resource_constructor

amqp_rpc_reply_t res = amqp_login_with_properties(
		resource->connection_state,
		params->vhost,
		params->channel_max,
		params->frame_max,
		params->heartbeat,
		&custom_properties_table,
==>		AMQP_SASL_METHOD_PLAIN, <==
		params->login,
		params->password
	);

Doing so, it is impossible to authenticate using ssl certificates, as these are requiring the sasl method AMQP_SASL_METHOD_EXTERNAL.
I have added a way to specify this method having an additional property sasl_method.
The external method enables php-amqp to authenticate using ssl, without the need of a password, instead using either the DN or CN of the certificate as the user name and by validating the certificate.
I am using this sucessfully on our system in a docker environment and would like to see this being part of the main php-ampq library.

pinepain commented on amqp.c:
```diff
@@ -172,6 +173,8 @@ static PHP_MINIT_FUNCTION(amqp) /* {{{ */
 	REGISTER_LONG_CONSTANT("AMQP_OS_SOCKET_TIMEOUT_ERRNO",	AMQP_OS_SOCKET_TIMEOUT_ERRNO,	CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("PHP_AMQP_MAX_CHANNELS",			PHP_AMQP_MAX_CHANNELS,			CONST_CS | CONST_PERSISTENT);
 
+    REGISTER_LONG_CONSTANT("AMQP_SASL_METHOD_PLAIN",		AMQP_SASL_METHOD_PLAIN,		CONST_CS | CONST_PERSISTENT);
```
formatting: tabs vs spaces, as this project uses tabs, for consistency purpose let's keep using tabs too.

BTW, I'm having this issue all the time too =).

pinepain commented on amqp_connection.c:
```diff
@@ -1238,6 +1248,56 @@ static PHP_METHOD(amqp_connection_class, setVerify)
+	int method;
+
+	/* Get the port from the method params */
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "z", &zvalMethod) == FAILURE) {
```
Let's just enforce it to be decimal and all conversion will be done by PHP

pinepain commented:
@bhoehl Thanks for contribution, looks decent. After minor cleanup and more eyes I'm more then happy to have it in the core too!

bhoehl commented:
@pinepain 
I have addressed the issues with tabs and spaces
also I have looked for the unit tests, but for some reason my commit didn´t fix it
is there a way to get a bit more details on those failures ?

pinepain commented:
@bhoehl thanks! Give another comment regards regards long a look (I see that you take `setPort` as a reference, but it also should be refactored, but later).

Regards broken tests, see travis build output (the red cross near the commit will get you there or just go to, say, https://travis-ci.org/pdezwart/php-amqp/builds/385629994)
